homeassistant/components/fully_kiosk/services.py

```diff
@@ -48,9 +49,11 @@ async def execute_service(
             if device:
                 for key in device.config_entries:
                     entry = hass.config_entries.async_get_entry(key)
-                    if not entry:
-                        continue
-                    if entry.domain != DOMAIN:
+                    if (
+                        not entry
+                        or entry.domain != DOMAIN
+                        or entry.state != ConfigEntryState.LOADED
+                    ):
                         continue
```

Comment:
If we don't find any matching coordinators, ie config entries, for the device at all, we should raise `HomeAssistantError` at the end of the function. This will notify the user that something went wrong with the service call.

Reply:
Thanks, I was going to ask about that. Doing nothing didn't seem like the right thing to do but it looks like a lot of integrations fail silently.

Reply:
@MartinHjelmare I have refactored how services get invoked by the Fully Kiosk Browser integration so that invalid device ids provided by the user will result in a `HomeAssistantError`. If the device's config entry is not loaded, a warning will be logged. I have added tests that verify the `HomeAssistantError` is getting raised as well as ensuring that an unloaded config entry will not be used.

Any feedback would be appreciated.